Unit/Unit.cpp

```diff
@@ -0,0 +1,80 @@
+#include <iostream>
+#include "Unit.h"
+
+using namespace std;
+
+void Unit::ensureIsAlive() {
+	if ( hitPoints == 0 ) {
+		throw UnitIsDeadException();
+	}
+}
+
+Unit::Unit(const std::string& name, int hp, int dmg) {
+	this->name = name;
+	this->hitPoints = hp;
+	this->hitPointsLimit = hp;
+	this->damage = dmg;
+}
+
+Unit::~Unit() {}
+
+int Unit::getDamage() const {
+	return damage;
+}
+
+int Unit::getHitPoints() const {
+	return hitPoints;
+}
+
+int Unit::getHitPointsLimit() const {
+	return hitPointsLimit;
+}
+
+const std::string& Unit::getName() const {
+	return name;
+}
+
+void Unit::addHitPoints(int hp) {
+	ensureIsAlive();
+
+	if ( hitPointsLimit - hitPoints < hp ) {
+		hitPoints = hitPointsLimit;
+
+		return;
+	}
+	hitPoints += hp;
+}
+
+void Unit::takeDamage(int dmg) {
+	hitPoints -= dmg;
+	if ( hitPoints <= 0 ) {
+		hitPoints = 0;
+	}
+}
+
+void Unit::attack(Unit& enemy) {
+	ensureIsAlive();
+	enemy.ensureIsAlive();
```

Comment:
It's better to put this in "take damage". Handle exception thrown by takeDamage in current method